sotodlib/tod_ops/jumps.py

```
@@ -182,33 +177,39 @@ def jumpfix_subtract_heights(
         x_fixed: x with jumps removed.
                  If inplace is True this is just a reference to x.
     """
-
-    def _fix(jump_ranges, heights, x_fixed):
-        for j, jump_range in enumerate(jump_ranges):
-            for start, end in jump_range.ranges():
-                _heights = heights[j, start:end]
-                height = _heights[np.argmax(np.abs(_heights))]
-                x_fixed[j, int((start + end) / 2) :] -= height
-
+    orig_shape = x.shape
+    x = np.atleast_2d(x)
+    x = np.ascontiguousarray(x)
```

If user requested inplace, and x is not contiguous, then the change won't be in-place.  So you need to check for that.

For example
```
x = np.asarray(x, order='C', copy=(False if inplace else None))
```

---

Do we really want to raise a value error if inplace is requested but the array is not contiguous?
I could instead raise a warning, make a copy, and then copy back into the original `x` later. But that may be babying the user too much.

---

Actually assuming we want to raise the error in that case is the most correct block something like:

```
x_fixed = np.asarray(np.atleast_2d(x), order="C", copy=(False if inplace else None))
```

That way the cases are as follows:

* in place and already contiguous -> `x_fixed` just references `x`
* in place and not contiguous -> `ValueError`
* no in place and already contiguous -> `x_fixed` is a copy of `x`
* no in place and not contiguous -> `x_fixed` is a contiguous copy of `x`

Does that make sense?

---

W.r.t. "no in place and already contiguous -> x_fixed is a copy of x" -- actually not true; `np.asarray(... copy=None)` will not copy if it doesn't need to.

Another way to approach this, to minimize copies and simplify logic:
- Apply all the conversions you need to get `x_fixed`, minimizing copies where possible.
- *Check* whether you ended up making a copy, or not -- `fixed_is_a_copy = np.may_share_memory(x, x_fixed)`
- Take different actions depending on `(fixed_is_a_copy, inplace)`.

I'm ok with your suggestion of making a copy and then updating x at the end.

The reason to ValueError would be that "inplace" is sometimes used not just to compress code, but to minimize copying of the data -- it's sort of implied that doing things inplace will use less RAM, less time.  But I think printing a warning is sufficient to help super-optimizers tweak things up.